pkg/cloudprovider/aws/cloudprovider.go

```diff
@@ -132,12 +125,18 @@ func (c *CloudProvider) create(ctx context.Context, constraints *v1alpha5.Constr
 	return errs
 }
 
+// Despite accepting a Constraints struct, note that it does not utilize Requirements at all and instead
+// returns all available InstanceTypes.
 func (c *CloudProvider) GetInstanceTypes(ctx context.Context, constraints *v1alpha5.Constraints) ([]cloudprovider.InstanceType, error) {
 	vendorConstraints, err := v1alpha1.Deserialize(constraints)
 	if err != nil {
 		return nil, apis.ErrGeneric(err.Error())
 	}
-	return c.instanceTypeProvider.Get(ctx, vendorConstraints)
+	allInstanceTypes, err := c.instanceTypeProvider.Get(ctx, vendorConstraints)
+	if err != nil {
+		return nil, err
+	}
+	return c.instanceProvider.DiscardICEdInstanceTypes(ctx, allInstanceTypes), nil
```

Comment:
Given that this is the only place that this is used, does it make sense to hide this behavior inside the instance provider? That is to say, instanceTypeProvider.Get() inherently factors in unavailable offerings. Is there ever a case where we want to include unavailable offerings on this call?